src/Reflection/ReflectionHelper.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\Larastan\Reflection;
+
+use PHPStan\PhpDoc\Tag\PropertyTag;
+use PHPStan\Reflection\ClassReflection;
+
+final class ReflectionHelper
+{
+    /**
+     * Returns all property tags of the given class and its ancestors.
+     *
+     * In case of duplicates, the tags of the class itself have precedence.
+     * TODO consider the hierarchy to ensure correct precedence between ancestors
+     *
+     * @return array<string, PropertyTag>
+     */
+    public static function collectPropertyTags(ClassReflection $classReflection): array
+    {
+        $allPropertyTags = $classReflection->getPropertyTags();
+
+        foreach ($classReflection->getAncestors() as $ancestor) {
+            $allPropertyTags += $ancestor->getPropertyTags();
+        }
+
+        return $allPropertyTags;
+    }
```

Comment:
```suggestion
    public static function hasPropertyTagInAncestors(ClassReflection $classReflection, string $propertyName): bool
    {
        foreach ($classReflection->getAncestors() as $ancestor) {
            if (array_key_exists($propertyName, $ancestor->getPropertyTags())) {
                return true;
            }
        }

        return false;
    }
```

I'd prefer something like this. So we don't need to collect every property tag. We can return early. And of course the calling code should be changed accordingly.

And I think we shouldn't be concerned too much about the hierarchy. Defining the same property with different types is rare.

Reply:
Done: [`6f84347` (#1037)](https://github.com/nunomaduro/larastan/pull/1037/commits/6f84347628ac5855da2959d4090710928e1ed856)

We don't have to worry about the hierarchy with this approach, since the method now only cares if the property is defined anywhere at all.